### ISSUE TITLE
setProgress crash

### ISSUE THREAD

**Issue by romantiskt** [airbnb/lottie-android#1063]
genymotion simulator -5.1.0 -API 22   768*1280
Below is my lottie:2.5.5 code ,which works properly on **2.5.5** 

```
LottieComposition.Factory
                    .fromInputStream(inputStream, new OnCompositionLoadedListener() {
                        @Override
                        public void onCompositionLoaded(LottieComposition composition) {
                            if (animationView != null && composition != null) {
                                try {
                                    animationView.setComposition(composition);
                                } catch (Exception e) {
                                    e.printStackTrace();
                                }
                            }
                        }
                    });

            animationView.setImageAssetDelegate(new ImageAssetDelegate() {
                @Override
                public Bitmap fetchBitmap(LottieImageAsset asset) {
                    String imagePath = presenter.getImagesPath(mContext, srcUrl)
                            + File.separator + asset.getFileName();
                    BitmapFactory.Options options = new BitmapFactory.Options();
                    options.inJustDecodeBounds = false;
                    options.inPreferredConfig = Bitmap.Config.RGB_565;
                    options.inDither = true;
                    return BitmapFactory.decodeFile(imagePath, options);
                }
            });
```

When I update to **2.7.0** and change it to the following code, it will generate an exception on the emulator, but it will work properly on the real machine
```
LottieTask<LottieComposition> lottieCompositionLottieTask = LottieCompositionFactory.fromJsonInputStream(inputStream, url);
            lottieCompositionLottieTask.addListener(new LottieListener<LottieComposition>() {
                @Override
                public void onResult(LottieComposition composition) {
                    if (animationView != null && composition != null) {
                        try {
                            animationView.setComposition(composition);
                        } catch (Exception e) {
                            e.printStackTrace();
                        }
                    }
                }
            });
            animationView.setImageAssetDelegate(new ImageAssetDelegate() {
                @Override
                public Bitmap fetchBitmap(LottieImageAsset asset) {
                    String imagePath = presenter.getImagesPath(mContext, srcUrl)
                            + File.separator + asset.getFileName();
                    BitmapFactory.Options options = new BitmapFactory.Options();
                    // 不用加载图片返回图片
                    options.inJustDecodeBounds = false;
                    options.inPreferredConfig = Bitmap.Config.RGB_565;
                    options.inDither = true;
                    return BitmapFactory.decodeFile(imagePath, options);
                }
            });
```
The detailed log of the exception generation is as follows

```
Tried to mark 0x18 not contained by any spaces                                                                                                                                       
Attempting see if it's a bad root                                                                                                                                                    
Found invalid root: 0x18                                                                                                                                                             
Type=RootJavaFrame thread_id=1 location=Visiting method 'void com.airbnb.lottie.model.layer.BaseLayer.setProgress(float)' at dex PC 0x0060 (native PC 0xe33353e0) vreg=2             
art/runtime/gc/collector/mark_sweep.cc:381] Can't mark invalid object                                                                                                                
art/runtime/runtime.cc:289] Runtime aborting...                                                                                                                                      
art/runtime/runtime.cc:289] Aborting thread:                                                                                                                                         
art/runtime/runtime.cc:289] "main" prio=5 tid=1 Runnable                                                                                                                             
art/runtime/runtime.cc:289]   | group="" sCount=0 dsCount=0 obj=0x7491e000 self=0xf3c25800                                                                                           
art/runtime/runtime.cc:289]   | sysTid=30200 nice=0 cgrp=default sched=0/0 handle=0xf7780ea0                                                                                         
art/runtime/runtime.cc:289]   | state=R schedstat=( 2144748775 171786360 1870 ) utm=156 stm=57 core=1 HZ=100                                                                         
art/runtime/runtime.cc:289]   | stack=0xff418000-0xff41a000 stackSize=8MB                                                                                                            
art/runtime/runtime.cc:289]   | held mutexes= "abort lock" "mutator lock"(shared held)                                                                                               
art/runtime/runtime.cc:289]   native: #00 pc 00005d03  /system/lib/libbacktrace_libc++.so (UnwindCurrent::Unwind(unsigned int, ucontext*)+83)                                        
art/runtime/runtime.cc:289]   native: #01 pc 00003051  /system/lib/libbacktrace_libc++.so (Backtrace::Unwind(unsigned int, ucontext*)+33)                                            
art/runtime/runtime.cc:289]   native: #02 pc 003c9b97  /system/lib/libart.so (art::DumpNativeStack(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, int, char const*, ar
art/runtime/runtime.cc:289]   native: #03 pc 00392f52  /system/lib/libart.so (art::Thread::Dump(std::__1::basic_ostream<char, std::__1::char_traits<char> >&) const+306)             
art/runtime/runtime.cc:289]   native: #04 pc 00377674  /system/lib/libart.so (art::AbortState::Dump(std::__1::basic_ostream<char, std::__1::char_traits<char> >&)+644)               
art/runtime/runtime.cc:289]   native: #05 pc 00377bdd  /system/lib/libart.so (art::Runtime::Abort()+125)                                                                             
art/runtime/runtime.cc:289]   native: #06 pc 000ec704  /system/lib/libart.so (art::LogMessage::~LogMessage()+1876)                                                                   
art/runtime/runtime.cc:289]   native: #07 pc 001cb8d1  /system/lib/libart.so (bool art::gc::accounting::HeapBitmap::AtomicTestAndSet<art::gc::collector::MarkSweepMarkObjectSlowPath>
   native: #08 pc 00084e4a  /system/lib/libart.so (art::gc::collector::MarkSweep::MarkObjectParallel(art::mirror::Object const*) (.part.121)+38)                                
   native: #09 pc 001cf74a  /system/lib/libart.so (art::gc::collector::MarkSweep::MarkRootParallelCallback(art::mirror::Object**, void*, art::RootInfo const&)+138)             
   native: #10 pc 00399d05  /system/lib/libart.so (art::ReferenceMapVisitor<art::RootCallbackVisitor>::VisitQuickFrame()+1269)                                                  
   native: #11 pc 0039a25f  /system/lib/libart.so (art::ReferenceMapVisitor<art::RootCallbackVisitor>::VisitFrame()+319)                                                        
   native: #12 pc 00384977  /system/lib/libart.so (art::StackVisitor::WalkStack(bool)+151)                                                                                      
   native: #13 pc 0038a070  /system/lib/libart.so (art::Thread::VisitRoots(void (*)(art::mirror::Object**, void*, art::RootInfo const&), void*)+1744)                           
   native: #14 pc 001cbca9  /system/lib/libart.so (art::gc::collector::CheckpointMarkThreadRoots::Run(art::Thread*)+137)                                                        
   native: #15 pc 0038a959  /system/lib/libart.so (art::Thread::RunCheckpointFunction()+185)                                                                                    
   native: #16 pc 0042e1ec  /system/lib/libart.so (artTestSuspendFromCode+92)                                                                                                   
   native: #17 pc 000e5b52  /system/lib/libart.so (art_quick_test_suspend+34)                                                                                                   
   native: #18 pc 0003615e  /data/dalvik-cache/x86/system@framework@boot.oat (Java_java_util_ArrayList_isEmpty__+66)                                                            
   at java.util.ArrayList.isEmpty(ArrayList.java:323)                                                                                                                           
   at com.airbnb.lottie.animation.keyframe.BaseKeyframeAnimation.getEndProgress(BaseKeyframeAnimation.java:121)                                                                 
   at com.airbnb.lottie.animation.keyframe.BaseKeyframeAnimation.setProgress(BaseKeyframeAnimation.java:46)                                                                     
   at com.airbnb.lottie.model.layer.BaseLayer.setProgress(BaseLayer.java:427)                                                                                                   
   at com.airbnb.lottie.model.layer.CompositionLayer.setProgress(CompositionLayer.java:139)                                                                                     
   at com.airbnb.lottie.model.layer.CompositionLayer.setProgress(CompositionLayer.java:139)                                                                                     
   at com.airbnb.lottie.LottieDrawable$1.onAnimationUpdate(LottieDrawable.java:103)                                                                                             
   at com.airbnb.lottie.utils.BaseLottieAnimator.notifyUpdate(BaseLottieAnimator.java:88)                                                                                       
   at com.airbnb.lottie.utils.LottieValueAnimator.doFrame(LottieValueAnimator.java:96)                                                                                          
   at android.view.Choreographer$CallbackRecord.run(Choreographer.java:765)                                                                                                     
   at android.view.Choreographer.doCallbacks(Choreographer.java:580)                                                                                                            
   at android.view.Choreographer.doFrame(Choreographer.java:549)                                                                                                                
   at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:753)                                                                                          
   at android.os.Handler.handleCallback(Handler.java:739)                                                                                                                       
   at android.os.Handler.dispatchMessage(Handler.java:95)                                                                                                                       
   at android.os.Looper.loop(Looper.java:135)                                                                                                                                   
   at android.app.ActivityThread.main(ActivityThread.java:5254)                                                                                                                 
   at java.lang.reflect.Method.invoke!(Native method)                                                                                                                           
   at java.lang.reflect.Method.invoke(Method.java:372)                                                                                                                          
   at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)                                                                                           
   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)                                                                                                              
 Dumping all threads without appropriate locks held: thread list lock                                                                                                           
 All threads:                                                                                                                                                                   
 DALVIK THREADS (22):                                                                                                                                                           
 "main" prio=5 tid=1 Runnable                                                                                                                                                   
   | group="" sCount=0 dsCount=0 obj=0x7491e000 self=0xf3c25800                                                                                                                 
   | sysTid=30200 nice=0 cgrp=default sched=0/0 handle=0xf7780ea0                                                                                                               
   | state=R schedstat=( 2145666414 171910472 1871 ) utm=156 stm=57 core=1 HZ=100                                                                                               
   | stack=0xff418000-0xff41a000 stackSize=8MB                                                                                                                                  
   | held mutexes= "abort lock" "mutator lock"(shared held)                                                                                                                     
   native: #00 pc 00005d03  /system/lib/libbacktrace_libc++.so (UnwindCurrent::Unwind(unsigned int, ucontext*)+83)                                                              
   native: #01 pc 00003051  /system/lib/libbacktrace_libc++.so (Backtrace::Unwind(unsigned int, ucontext*)+33)                                                                  
   native: #02 pc 003c9b97  /system/lib/libart.so (art::DumpNativeStack(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, int, char const*, art::mirror::ArtMethod*)
   native: #03 pc 00392f52  /system/lib/libart.so (art::Thread::Dump(std::__1::basic_ostream<char, std::__1::char_traits<char> >&) const+306)                                   
   native: #04 pc 0039f34e  /system/lib/libart.so (art::ThreadList::DumpLocked(std::__1::basic_ostream<char, std::__1::char_traits<char> >&)+478)                               
   native: #05 pc 003775ec  /system/lib/libart.so (art::AbortState::Dump(std::__1::basic_ostream<char, std::__1::char_traits<char> >&)+508)                                     
   native: #06 pc 00377bdd  /system/lib/libart.so (art::Runtime::Abort()+125)                                                                                                   
   native: #07 pc 000ec704  /system/lib/libart.so (art::LogMessage::~LogMessage()+1876)                                                                                         
   native: #08 pc 001cb8d1  /system/lib/libart.so (bool art::gc::accounting::HeapBitmap::AtomicTestAndSet<art::gc::collector::MarkSweepMarkObjectSlowPath>(art::mirror::Object c
   native: #09 pc 00084e4a  /system/lib/libart.so (art::gc::collector::MarkSweep::MarkObjectParallel(art::mirror::Object const*) (.part.121)+38)                                
   native: #10 pc 001cf74a  /system/lib/libart.so (art::gc::collector::MarkSweep::MarkRootParallelCallback(art::mirror::Object**, void*, art::RootInfo const&)+138)             
   native: #11 pc 00399d05  /system/lib/libart.so (art::ReferenceMapVisitor<art::RootCallbackVisitor>::VisitQuickFrame()+1269)                                                  
   native: #12 pc 0039a25f  /system/lib/libart.so (art::ReferenceMapVisitor<art::RootCallbackVisitor>::VisitFrame()+319)                                                        
   native: #13 pc 00384977  /system/lib/libart.so (art::StackVisitor::WalkStack(bool)+151)                                                                                      
   native: #14 pc 0038a070  /system/lib/libart.so (art::Thread::VisitRoots(void (*)(art::mirror::Object**, void*, art::RootInfo const&), void*)+1744)                           
   native: #15 pc 001cbca9  /system/lib/libart.so (art::gc::collector::CheckpointMarkThreadRoots::Run(art::Thread*)+137)                                                        
   native: #16 pc 0038a959  /system/lib/libart.so (art::Thread::RunCheckpointFunction()+185)                                                                                    
   native: #17 pc 0042e1ec  /system/lib/libart.so (artTestSuspendFromCode+92)                                                                                                   
   native: #18 pc 000e5b52  /system/lib/libart.so (art_quick_test_suspend+34)                                                                                                   
   native: #19 pc 0003615e  /data/dalvik-cache/x86/system@framework@boot.oat (Java_java_util_ArrayList_isEmpty__+66)                                                            
   at java.util.ArrayList.isEmpty(ArrayList.java:323)                                                                                                                           
   at com.airbnb.lottie.animation.keyframe.BaseKeyframeAnimation.getEndProgress(BaseKeyframeAnimation.java:121)                                                                 
   at com.airbnb.lottie.animation.keyframe.BaseKeyframeAnimation.setProgress(BaseKeyframeAnimation.java:46)                                                                     
   at com.airbnb.lottie.model.layer.BaseLayer.setProgress(BaseLayer.java:427)                                                                                                   
   at com.airbnb.lottie.model.layer.CompositionLayer.setProgress(CompositionLayer.java:139)                                                                                     
   at com.airbnb.lottie.model.layer.CompositionLayer.setProgress(CompositionLayer.java:139)                                                                                     
   at com.airbnb.lottie.LottieDrawable$1.onAnimationUpdate(LottieDrawable.java:103)                                                                                             
   at com.airbnb.lottie.utils.BaseLottieAnimator.notifyUpdate(BaseLottieAnimator.java:88)                                                                                       
   at com.airbnb.lottie.utils.LottieValueAnimator.doFrame(LottieValueAnimator.java:96)                                                                                          
   at android.view.Choreographer$CallbackRecord.run(Choreographer.java:765)                                                                                                     
   at android.view.Choreographer.doCallbacks(Choreographer.java:580)                                                                                                            
   at android.view.Choreographer.doFrame(Choreographer.java:549)                                                                                                                
   at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:753)                                                                                          
   at android.os.Handler.handleCallback(Handler.java:739)                                                                                                                       
   at android.os.Handler.dispatchMessage(Handler.java:95)                                                                                                                       
   at android.os.Looper.loop(Looper.java:135)                                                                                                                                   
   at android.app.ActivityThread.main(ActivityThread.java:5254)                                                                                                                 
   at java.lang.reflect.Method.invoke!(Native method)
 "Heap thread pool worker thread 1" prio=5 tid=2 Native (still starting up)                                                                                    
   | group="" sCount=0 dsCount=0 obj=0x0 self=0xeec62400                                                                                                       
   | sysTid=30207 nice=0 cgrp=default sched=0/0 handle=0xf3c51580                                                                                              
   | state=S schedstat=( 5536995 4376275 19 ) utm=0 stm=0 core=1 HZ=100                                                                                        
   | stack=0xf30f9000-0xf30fb000 stackSize=1020KB                                                                                                              
   | held mutexes=                                                                                                                                             
   kernel: futex_wait_queue_me+0xc0/0x110                                                                                                                      
   kernel: futex_wait+0x112/0x250                                                                                                                              
   kernel: do_futex+0xdc/0xb30                                                                                                                                 
   kernel: compat_SyS_futex+0x75/0x150                                                                                                                         
   kernel: do_syscall_32_irqs_off+0x5f/0x180                                                                                                                   
   kernel: entry_INT80_compat+0x36/0x50                                                                                                                        
   native: #00 pc 00013300  /system/lib/libc.so (syscall+32)                                                                                                   
   native: #01 pc 000fdc97  /dev/ashmem/dalvik-Heap thread pool worker thread 1 (deleted) (???)                                                                
   (no managed stack frames)                                                                                                                                   
                                                                                                                                                               
 "Heap thread pool worker thread 2" prio=5 tid=3 Native (still starting up)                                                                                    
   | group="" sCount=0 dsCount=0 obj=0x0 self=0xf3c25c00                                                                                                       
   | sysTid=30208 nice=0 cgrp=default sched=0/0 handle=0xf3c51800                                                                                              
   | state=S schedstat=( 5723251 4011593 16 ) utm=0 stm=0 core=1 HZ=100                                                                                        
   | stack=0xf2ff9000-0xf2ffb000 stackSize=1020KB                                                                                                              
   | held mutexes=                                                                                                                                             
   kernel: futex_wait_queue_me+0xc0/0x110                                                                                                                      
   kernel: futex_wait+0x112/0x250                                                                                                                              
   kernel: do_futex+0xdc/0xb30                                                                                                                                 
   kernel: compat_SyS_futex+0x75/0x150                                                                                                                         
   kernel: do_syscall_32_irqs_off+0x5f/0x180                                                                                                                   
   kernel: entry_INT80_compat+0x36/0x50                                                                                                                        
   native: #00 pc 00013300  /system/lib/libc.so (syscall+32)                                                                                                   
   native: #01 pc 000fdc97  /dev/ashmem/dalvik-Heap thread pool worker thread 2 (deleted) (???)                                                                
   (no managed stack frames)                                                                                                                                   
                                                                                                                                                               
 "Heap thread pool worker thread 0" prio=5 tid=4 Native (still starting up)                                                                                    
   | group="" sCount=0 dsCount=0 obj=0x0 self=0xeec63400                                                                                                       
   | sysTid=30206 nice=0 cgrp=default sched=0/0 handle=0xf3c53b00                                                                                              
   | state=S schedstat=( 8124391 6976817 18 ) utm=0 stm=0 core=3 HZ=100                                                                                        
   | stack=0xf31f9000-0xf31fb000 stackSize=1020KB                                                                                                              
   | held mutexes=                                                                                                                                             
   kernel: futex_wait_queue_me+0xc0/0x110                                                                                                                      
   kernel: futex_wait+0x112/0x250                                                                                                                              
   kernel: do_futex+0xdc/0xb30                                                                                                                                 
   kernel: compat_SyS_futex+0x75/0x150                                                                                                                         
   kernel: do_syscall_32_irqs_off+0x5f/0x180                                                                                                                   
   kernel: entry_INT80_compat+0x36/0x50                                                                                                                        
   native: #00 pc 00013300  /system/lib/libc.so (syscall+32)                                                                                                   
   native: #01 pc 000fdc97  /dev/ashmem/dalvik-Heap thread pool worker thread 0 (deleted) (???)                                                                
   (no managed stack frames)                                                                                                                                   
                                                                                                                                                               
 "Signal Catcher" prio=5 tid=5 WaitingInMainSignalCatcherLoop                                                                                                  
   | group="" sCount=0 dsCount=0 obj=0x12c000a0 self=0xf3c26c00                                                                                                
   | sysTid=30209 nice=0 cgrp=default sched=0/0 handle=0xf3c51a80                                                                                              
   | state=S schedstat=( 400284 612794 1 ) utm=0 stm=0 core=1 HZ=100                                                                                           
   | stack=0xf2efb000-0xf2efd000 stackSize=1012KB                                                                                                              
   | held mutexes=                                                                                                                                             
   kernel: do_sigtimedwait+0x152/0x1e0                                                                                                                         
   kernel: C_SYSC_rt_sigtimedwait+0x8d/0xe0                                                                                                                    
   kernel: compat_SyS_rt_sigtimedwait+0xe/0x10                                                                                                                 
   kernel: do_syscall_32_irqs_off+0x5f/0x180                                                                                                                   
   kernel: entry_INT80_compat+0x36/0x50                                                                                                                        
   native: #00 pc 0007498b  /system/lib/libc.so (__rt_sigtimedwait+27)                                                                                         
   native: #01 pc 00024397  /system/lib/libc.so (sigwait+71)                                                                                                   
   native: #02 pc 0037fd45  /system/lib/libart.so (art::SignalCatcher::WaitForSignal(art::Thread*, art::SignalSet&)+133)                                       
   native: #03 pc 00380962  /system/lib/libart.so (art::SignalCatcher::Run(void*)+306)                                                                         
   native: #04 pc 000211d8  /system/lib/libc.so (__pthread_start(void*)+56)                                                                                    
   native: #05 pc 0001c559  /system/lib/libc.so (__start_thread+25)                                                                                            
   native: #06 pc 00013126  /system/lib/libc.so (__bionic_clone+70)                                                                                            
   (no managed stack frames)                                                                                                                                   
                                                                                                                                                               
 "JDWP" prio=5 tid=6 WaitingInMainDebuggerLoop                                                                                                                 
   | group="" sCount=0 dsCount=0 obj=0x12c050a0 self=0xeec64400                                                                                                
   | sysTid=30210 nice=0 cgrp=default sched=0/0 handle=0xf3c51d00                                                                                              
   | state=S schedstat=( 2984449 270050 10 ) utm=0 stm=0 core=2 HZ=100                                                                                         
   | stack=0xf2dfd000-0xf2dff000 stackSize=1012KB                                                                                                              
   | held mutexes=                                                                                                                                             
   kernel: poll_schedule_timeout+0x49/0x70                                                                                                                     
   kernel: do_select+0x57d/0x7c0                                                                                                                               
   kernel: compat_core_sys_select+0x1cf/0x290                                                                                                                  
   kernel: compat_SyS_pselect6+0x21a/0x240                                                                                                                     
   kernel: do_syscall_32_irqs_off+0x5f/0x180                                                                                                                   
   kernel: entry_INT80_compat+0x36/0x50                                                                                                                        
   native: #00 pc 00074445  /system/lib/libc.so (__pselect6+37)                                                                                                
   native: #01 pc 0001fcf3  /system/lib/libc.so (select+179)                                                                                                   
   native: #02 pc 0043892f  /system/lib/libart.so (art::JDWP::JdwpAdbState::ProcessIncoming()+335)                                                             
   native: #03 pc 0028f4df  /system/lib/libart.so (art::JDWP::JdwpState::Run()+415)                                                                            
   native: #04 pc 002915d0  /system/lib/libart.so (art::JDWP::StartJdwpThread(void*)+32)                                                                       
   native: #05 pc 000211d8  /system/lib/libc.so (__pthread_start(void*)+56)                                                                                    
   native: #06 pc 0001c559  /system/lib/libc.so (__start_thread+25)                                                                                            
   native: #07 pc 00013126  /system/lib/libc.so (__bionic_clone+70)                                                                                            
   (no managed stack frames)                                                                                                                                   
                                                                                                                                                               
 "ReferenceQueueDaemon" prio=5 tid=7 Waiting                                                                                                                   
   | group="" sCount=0 dsCount=0 obj=0x12c0d0a0 self=0xf3c27000                                                                                                
   | sysTid=30211 nice=0 cgrp=default sched=0/0 handle=0xf3c51f80                                                                                              
   | state=S schedstat=( 1396305 547510 66 ) utm=0 stm=0 core=2 HZ=100                                                                                         
   | stack=0xf2cf9000-0xf2cfb000 stackSize=1036KB                                                                                                              
   | held mutexes=                                                                                                                                             
   kernel: futex_wait_queue_me+0xc0/0x110                                                                                                                      
   kernel: futex_wait+0x112/0x250                                                                                                                              
   kernel: do_futex+0xdc/0xb30                                                                                                                                 
   kernel: compat_SyS_futex+0x75/0x150                                                                                                                         
   kernel: do_syscall_32_irqs_off+0x5f/0x180                                                                                                                   
   kernel: entry_INT80_compat+0x36/0x50                                                                                                                        
   native: #00 pc 00013300  /system/lib/libc.so (syscall+32)                                                                                                   
   native: #01 pc 001018e7  [stack:30211] (???)                                                                                                                
   at java.lang.Object.wait!(Native method)                                                                                                                    
   - waiting on <0x00968fb8> (a java.lang.Class<java.lang.ref.ReferenceQueue>)                                                                                 
   at java.lang.Daemons$ReferenceQueueDaemon.run(Daemons.java:134)                                                                                             
   - locked <0x00968fb8> (a java.lang.Class<java.lang.ref.ReferenceQueue>)                                                                                     
   at java.lang.Thread.run(Thread.java:818)                                                                                                                    
                                                                                                                                                               
 "FinalizerDaemon" prio=5 tid=8 Waiting                                                                                                                        
   | group="" sCount=0 dsCount=0 obj=0x12c0d100 self=0xf3c31800                                                                                                
   | sysTid=30212 nice=0 cgrp=default sched=0/0 handle=0xf3c53d80                                                                                              
   | state=S schedstat=( 3330141 915940 16 ) utm=0 stm=0 core=2 HZ=100                                                                                         
   | stack=0xeeafe000-0xeeb00000 stackSize=1036KB                                                                                                              
   | held mutexes=                                                                                                                                             
   kernel: futex_wait_queue_me+0xc0/0x110                                                                                                                      
   kernel: futex_wait+0x112/0x250                                                                                                                              
   kernel: do_futex+0xdc/0xb30                                                                                                                                 
   kernel: compat_SyS_futex+0x75/0x150                                                                                                                         
   kernel: do_syscall_32_irqs_off+0x5f/0x180                                                                                                                   
   kernel: entry_INT80_compat+0x36/0x50                                                                                                                        
   native: #00 pc 00013300  /system/lib/libc.so (syscall+32)                                                                                                   
   native: #01 pc 00101807  [stack:30212] (???)                                                                                                                
   at java.lang.Object.wait!(Native method)                                                                                                                    
   - waiting on <0x00357191> (a java.lang.ref.ReferenceQueue)                                                                                                  
   at java.lang.Object.wait(Object.java:422)                                                                                                                   
   at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:101)                                                                                             
   - locked <0x00357191> (a java.lang.ref.ReferenceQueue)                                                                                                      
   at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:72)                                                                                              
   at java.lang.Daemons$FinalizerDaemon.run(Daemons.java:174)                                                                                                  
   at java.lang.Thread.run(Thread.java:818)                                                                                                                    
                                                                                                                                                               
 "FinalizerWatchdogDaemon" prio=5 tid=9 Waiting                                                                                                                
   | group="" sCount=0 dsCount=0 obj=0x12c0d160 self=0xf3c31c00                                                                                                
   | sysTid=30213 nice=0 cgrp=default sched=0/0 handle=0xf3c54000                                                                                              
   | state=S schedstat=( 1272889 2914474 16 ) utm=0 stm=0 core=2 HZ=100                                                                                        
   | stack=0xe3b09000-0xe3b0b000 stackSize=1036KB                                                                                                              
   | held mutexes=                                                                                                                                             
   kernel: futex_wait_queue_me+0xc0/0x110                                                                                                                      
   kernel: futex_wait+0x112/0x250                                                                                                                              
   kernel: do_futex+0xdc/0xb30                                                                                                                                 
   kernel: compat_SyS_futex+0x75/0x150                                                                                                                         
   kernel: do_syscall_32_irqs_off+0x5f/0x180                                                                                                                   
   kernel: entry_INT80_compat+0x36/0x50                                                                                                                        
   native: #00 pc 00013300  /system/lib/libc.so (syscall+32)                                                                                                   
   native: #01 pc 001018b7  [stack:30213] (???)                                                                                                                
   at java.lang.Object.wait!(Native method)                                                                                                                    
   - waiting on <0x0681b8f6> (a java.lang.Daemons$FinalizerWatchdogDaemon)                                                                                     
   at java.lang.Daemons$FinalizerWatchdogDaemon.waitForObject(Daemons.java:240)                                                                                
   - locked <0x0681b8f6> (a java.lang.Daemons$FinalizerWatchdogDaemon)                                                                                         
   at java.lang.Daemons$FinalizerWatchdogDaemon.run(Daemons.java:212)                                                                                          
   at java.lang.Thread.run(Thread.java:818)                                                                                                                    
                                                                                                                                                               
 "HeapTrimmerDaemon" prio=5 tid=10 Sleeping                                                                                                                    
   | group="" sCount=0 dsCount=0 obj=0x12c0d1c0 self=0xf3c32000                                                                                                
   | sysTid=30214 nice=0 cgrp=default sched=0/0 handle=0xf3c54280                                                                                              
   | state=S schedstat=( 1734462 1428161 12 ) utm=0 stm=0 core=2 HZ=100                                                                                        
   | stack=0xe3a05000-0xe3a07000 stackSize=1036KB                                                                                                              
   | held mutexes=                                                                                                                                             
   kernel: hrtimer_nanosleep+0xae/0x190                                                                                                                        
   kernel: compat_SyS_nanosleep+0xb4/0x130                                                                                                                     
   kernel: do_syscall_32_irqs_off+0x5f/0x180                                                                                                                   
   kernel: entry_INT80_compat+0x36/0x50                                                                                                                        
   native: #00 pc 00075df1  /system/lib/libc.so (nanosleep+17)                                                                                                 
   native: #01 pc 00043748  /system/lib/libc.so (usleep+72)                                                                                                    
   native: #02 pc 001faa2c  /system/lib/libart.so (art::gc::Heap::DoPendingTransitionOrTrim()+1052)                                                            
   native: #03 pc 00337884  /system/lib/libart.so (art::VMRuntime_trimHeap(_JNIEnv*, _jobject*)+36)                                                            
   native: #04 pc 000004ef  /data/dalvik-cache/x86/system@framework@boot.oat (Java_dalvik_system_VMRuntime_trimHeap__+115)                                     
   at dalvik.system.VMRuntime.trimHeap(Native method)                                                                                                          
   - sleeping on an unknown object                                                                                                                             
   at java.lang.Daemons$HeapTrimmerDaemon.run(Daemons.java:313)                                                                                                
   at java.lang.Thread.run(Thread.java:818)                                                                                                                    
                                                                                                                                                               
 "GCDaemon" prio=5 tid=11 WaitingForCheckPointsToRun                                                                                                           
   | group="" sCount=0 dsCount=0 obj=0x12c0d220 self=0xf3c32400                                                                                                
   | sysTid=30215 nice=0 cgrp=default sched=0/0 handle=0xf3c54500                                                                                              
   | state=S schedstat=( 139049791 29701296 213 ) utm=6 stm=6 core=2 HZ=100                                                                                    
   | stack=0xe3901000-0xe3903000 stackSize=1036KB                                                                                                              
   | held mutexes=                                                                                                                                             
   kernel: futex_wait_queue_me+0xc0/0x110                                                                                                                      
   kernel: futex_wait+0x112/0x250                                                                                                                              
   kernel: do_futex+0xdc/0xb30                                                                                                                                 
   kernel: compat_SyS_futex+0x75/0x150                                                                                                                         
   kernel: do_syscall_32_irqs_off+0x5f/0x180                                                                                                                   
   kernel: entry_INT80_compat+0x36/0x50                                                                                                                        
   native: #00 pc 00013300  /system/lib/libc.so (syscall+32)                                                                                                   
   native: #01 pc 001011b7  [stack:30215] (???)                                                                                                                
   at dalvik.system.VMRuntime.concurrentGC(Native method)                                                                                                      
   at java.lang.Daemons$GCDaemon.run(Daemons.java:346)                                                                                                         
   at java.lang.Thread.run(Thread.java:818)                                                                                                                    
                                                                                                                                                               
 "Binder_1" prio=5 tid=12 Native                                                                                                                               
   | group="" sCount=0 dsCount=0 obj=0x12c540a0 self=0xeec65400                                                                                                
   | sysTid=30216 nice=0 cgrp=default sched=0/0 handle=0xf3c54780                                                                                              
   | state=S schedstat=( 10631381 12148778 127 ) utm=0 stm=0 core=1 HZ=100                                                                                     
   | stack=0xe3705000-0xe3707000 stackSize=1012KB                                                                                                              
   | held mutexes=                                                                                                                                             
   kernel: binder_thread_read+0x837/0xf00                                                                                                                      
   kernel: binder_ioctl_write_read.isra.39+0xf2/0x260                                                                                                          
   kernel: binder_ioctl+0x219/0x5ea                                                                                                                            
   kernel: compat_SyS_ioctl+0xc0/0x1130                                                                                                                        
   kernel: do_syscall_32_irqs_off+0x5f/0x180                                                                                                                   
   kernel: entry_INT80_compat+0x36/0x50                                                                                                                        
   native: #00 pc 00073e26  /system/lib/libc.so (__ioctl+22)                                                                                                   
   native: #01 pc 000adc5a  /system/lib/libc.so (ioctl+42)                                                                                                     
   native: #02 pc 0002e26b  /system/lib/libbinder.so (android::IPCThreadState::talkWithDriver(bool)+251)                                                       
   native: #03 pc 0002ec65  /system/lib/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+37)                                                      
   native: #04 pc 0002ed29  /system/lib/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+105)                                                       
   native: #05 pc 000377a8  /system/lib/libbinder.so (android::PoolThread::threadLoop()+40)                                                                    
   native: #06 pc 000169de  /system/lib/libutils.so (android::Thread::_threadLoop(void*)+398)                                                                  
   native: #07 pc 00071d92  /system/lib/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+98)                                             
   native: #08 pc 000160fa  /system/lib/libutils.so (thread_data_t::trampoline(thread_data_t const*)+122)                                                      
   native: #09 pc 000211d8  /system/lib/libc.so (__pthread_start(void*)+56)                                                                                    
   native: #10 pc 0001c559  /system/lib/libc.so (__start_thread+25)                                                                                            
   native: #11 pc 00013126  /system/lib/libc.so (__bionic_clone+70)                                                                                            
   (no managed stack frames)                                                                                                                                   
                                                                                                                                                               
 "Binder_2" prio=5 tid=13 Native                                                                                                                               
   | group="" sCount=0 dsCount=0 obj=0x12c680a0 self=0xf3c33800                                                                                                
   | sysTid=30217 nice=0 cgrp=default sched=0/0 handle=0xeeca9080                                                                                              
   | state=S schedstat=( 7838303 11653544 116 ) utm=0 stm=0 core=0 HZ=100                                                                                      
   | stack=0xe3607000-0xe3609000 stackSize=1012KB                                                                                                              
   | held mutexes=                                                                                                                                             
   kernel: binder_thread_read+0x837/0xf00                                                                                                                      
   kernel: binder_ioctl_write_read.isra.39+0xf2/0x260                                                                                                          
   kernel: binder_ioctl+0x219/0x5ea                                                                                                                            
   kernel: compat_SyS_ioctl+0xc0/0x1130                                                                                                                        
   kernel: do_syscall_32_irqs_off+0x5f/0x180                                                                                                                   
   kernel: entry_INT80_compat+0x36/0x50                                                                                                                        
   native: #00 pc 00073e26  /system/lib/libc.so (__ioctl+22)                                                                                                   
   native: #01 pc 000adc5a  /system/lib/libc.so (ioctl+42)                                                                                                     
   native: #02 pc 0002e26b  /system/lib/libbinder.so (android::IPCThreadState::talkWithDriver(bool)+251)                                                       
   native: #03 pc 0002ec65  /system/lib/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+37)                                                      
   native: #04 pc 0002ed29  /system/lib/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+105)                                                       
   native: #05 pc 000377a8  /system/lib/libbinder.so (android::PoolThread::threadLoop()+40)                                                                    
   native: #06 pc 000169de  /system/lib/libutils.so (android::Thread::_threadLoop(void*)+398)                                                                  
   native: #07 pc 00071d92  /system/lib/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+98)                                             
   native: #08 pc 000160fa  /system/lib/libutils.so (thread_data_t::trampoline(thread_data_t const*)+122)                                                      
   native: #09 pc 000211d8  /system/lib/libc.so (__pthread_start(void*)+56)                                                                                    
   native: #10 pc 0001c559  /system/lib/libc.so (__start_thread+25)                                                                                            
   native: #11 pc 00013126  /system/lib/libc.so (__bionic_clone+70)                                                                                            
   (no managed stack frames)                                                                                                                                   
                                                                                                                                                               
 "RenderThread" prio=5 tid=14 Native                                                                                                                           
   | group="" sCount=0 dsCount=0 obj=0x12c001c0 self=0xeec66400                                                                                                
   | sysTid=30220 nice=-4 cgrp=default sched=0/0 handle=0xf3c54a00                                                                                             
   | state=S schedstat=( 320325609 91521845 2136 ) utm=6 stm=25 core=2 HZ=100                                                                                  
   | stack=0xe1f42000-0xe1f44000 stackSize=1012KB                                                                                                              
   | held mutexes=                                                                                                                                             
   kernel: ep_poll+0x307/0x400                                                                                                                                 
   kernel: SyS_epoll_pwait+0x1c0/0x1f0                                                                                                                         
   kernel: do_syscall_32_irqs_off+0x5f/0x180                                                                                                                   
   kernel: entry_INT80_compat+0x36/0x50                                                                                                                        
   native: #00 pc 00073e65  /system/lib/libc.so (__epoll_pwait+37)                                                                                             
   native: #01 pc 0001cca2  /system/lib/libc.so (epoll_pwait+98)                                                                                               
   native: #02 pc 0001ccea  /system/lib/libc.so (epoll_wait+58)                                                                                                
   native: #03 pc 0001b6a7  /system/lib/libutils.so (android::Looper::pollInner(int)+199)                                                                      
   native: #04 pc 0001baa7  /system/lib/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+55)                                                    
   native: #05 pc 00089ca4  /system/lib/libhwui.so (android::uirenderer::renderthread::RenderThread::threadLoop()+116)                                         
   native: #06 pc 000169de  /system/lib/libutils.so (android::Thread::_threadLoop(void*)+398)                                                                  
   native: #07 pc 00071d92  /system/lib/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+98)                                             
   native: #08 pc 000160fa  /system/lib/libutils.so (thread_data_t::trampoline(thread_data_t const*)+122)                                                      
   native: #09 pc 000211d8  /system/lib/libc.so (__pthread_start(void*)+56)                                                                                    
   native: #10 pc 0001c559  /system/lib/libc.so (__start_thread+25)                                                                                            
   native: #11 pc 00013126  /system/lib/libc.so (__bionic_clone+70)                                                                                            
   (no managed stack frames)                                                                                                                                   
                                                                                                                                                               
 "hwuiTask1" prio=5 tid=15 Native                                                                                                                              
   | group="" sCount=0 dsCount=0 obj=0x12e580a0 self=0xf3f73c00                                                                                                
   | sysTid=30223 nice=-2 cgrp=default sched=0/0 handle=0xeeca9a80                                                                                             
   | state=S schedstat=( 62208350 39319518 355 ) utm=1 stm=4 core=1 HZ=100                                                                                     
   | stack=0xe1d8d000-0xe1d8f000 stackSize=1012KB                                                                                                              
   | held mutexes=                                                                                                                                             
   kernel: futex_wait_queue_me+0xc0/0x110                                                                                                                      
   kernel: futex_wait+0x112/0x250                                                                                                                              
   kernel: do_futex+0xdc/0xb30                                                                                                                                 
   kernel: compat_SyS_futex+0x75/0x150                                                                                                                         
   kernel: do_syscall_32_irqs_off+0x5f/0x180                                                                                                                   
   kernel: entry_INT80_compat+0x36/0x50                                                                                                                        
   native: #00 pc 00013300  /system/lib/libc.so (syscall+32)                                                                                                   
   native: #01 pc 000fbdb7  [stack:30223] (???)                                                                                                                
   (no managed stack frames)                                                                                                                                   
                                                                                                                                                               
 "hwuiTask2" prio=5 tid=16 Native                                                                                                                              
   | group="" sCount=0 dsCount=0 obj=0x12e5c0a0 self=0xeec67400                                                                                                
   | sysTid=30224 nice=-2 cgrp=default sched=0/0 handle=0xeeca9d00                                                                                             
   | state=S schedstat=( 14221558 11710769 39 ) utm=0 stm=1 core=3 HZ=100                                                                                      
   | stack=0xe1c8b000-0xe1c8d000 stackSize=1012KB                                                                                                              
   | held mutexes=                                                                                                                                             
   kernel: futex_wait_queue_me+0xc0/0x110                                                                                                                      
   kernel: futex_wait+0x112/0x250                                                                                                                              
   kernel: do_futex+0xdc/0xb30                                                                                                                                 
   kernel: compat_SyS_futex+0x75/0x150                                                                                                                         
   kernel: do_syscall_32_irqs_off+0x5f/0x180                                                                                                                   
   kernel: entry_INT80_compat+0x36/0x50                                                                                                                        
   native: #00 pc 00013300  /system/lib/libc.so (syscall+32)                                                                                                   
   native: #01 pc 000fbdb7  [stack:30224] (???)                                                                                                                
                                                                                                                                                               
   | group="" sCount=0 dsCount=0 obj=0x12e70a60 self=0xf3f75c00                                                                                                
   | state=S schedstat=( 2460763 26584 6 ) utm=0 stm=0 core=0 HZ=100                                                                                           
   | stack=0xe0efe000-0xe0f00000 stackSize=1036KB                                                                                                              
   kernel: futex_wait_queue_me+0xc0/0x110                                                                                                                      
   kernel: do_futex+0xdc/0xb30                                                                                                                                 
   kernel: do_syscall_32_irqs_off+0x5f/0x180                                                                                                                   
   native: #00 pc 00013302  /system/lib/libc.so (syscall+34)                                                                                                   
   at java.lang.Object.wait!(Native method)                                                                                                                    
   at java.lang.Thread.parkFor(Thread.java:1220)                                                                                                               
   at sun.misc.Unsafe.park(Unsafe.java:299)                                                                                                                    
   at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2055)                                   
   at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:776)                                             
   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1097)                                                                          
   at java.lang.Thread.run(Thread.java:818)                                                                                                                    
 "RxCachedWorkerPoolEvictor-1" prio=5 tid=18 TimedWaiting                                                                                                      
   | sysTid=30538 nice=0 cgrp=default sched=0/0 handle=0xf3c55e00                                                                                              
   | held mutexes=                                                                                                                                             
   kernel: futex_wait+0x112/0x250                                                                                                                              
   kernel: compat_SyS_futex+0x75/0x150                                                                                                                         
   kernel: entry_INT80_compat+0x36/0x50                                                                                                                        
   native: #01 pc 00101697  [stack:30538] (???)                                                                                                                
   at java.lang.Thread.parkFor(Thread.java:1220)                                                                                                               
   at sun.misc.Unsafe.park(Unsafe.java:299)                                                                                                                    
   at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2055)                                   
   at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:776)                                             
   at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1035)                                                                            
   at java.lang.Thread.run(Thread.java:818)                                                                                                                    
 "RxCachedThreadScheduler-1" prio=5 tid=19 Waiting                                                                                                             
   | sysTid=30539 nice=0 cgrp=default sched=0/0 handle=0xf3c56a80                                                                                              
   | stack=0xe0cf6000-0xe0cf8000 stackSize=1036KB                                                                                                              
   kernel: futex_wait_queue_me+0xc0/0x110                                                                                                                      
   kernel: do_futex+0xdc/0xb30                                                                                                                                 
   kernel: do_syscall_32_irqs_off+0x5f/0x180                                                                                                                   
   native: #00 pc 00013300  /system/lib/libc.so (syscall+32)                                                                                                   
   at java.lang.Object.wait!(Native method)                                                                                                                    
   at java.lang.Thread.parkFor(Thread.java:1220)                                                                                                               
   at sun.misc.Unsafe.park(Unsafe.java:299)                                                                                                                    
   at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2016)                                        
   at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:776)                                             
   at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1035)                                                                            
   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)                                                                          
                                                                                                                                                               
   | group="" sCount=0 dsCount=0 obj=0x12de0f40 self=0xf3f78000                                                                                                
   | state=S schedstat=( 48854780 10589091 45 ) utm=2 stm=2 core=1 HZ=100                                                                                      
   | held mutexes=                                                                                                                                             
   kernel: futex_wait+0x112/0x250                                                                                                                              
   kernel: compat_SyS_futex+0x75/0x150                                                                                                                         
   kernel: entry_INT80_compat+0x36/0x50                                                                                                                        
   native: #01 pc 001016b7  [stack:30540] (???)                                                                                                                                                                                                                                                                                           
```

**Comment by gpeal** [comment]
Duplicate of #1000
